fsm-calculator/src/main/java/org/egov/fsm/calculator/services/DemandService.java

```diff
@@ -0,0 +1,346 @@
+                        .build());
+            });
+
+
+//             addRoundOffTaxHead(calculation.getTenantId(),demandDetails);
+
+            int year = Calendar.getInstance().get(Calendar.YEAR);
+            Calendar startCal = Calendar.getInstance();
+            startCal.set(Calendar.YEAR,year);
+            startCal.set(Calendar.DAY_OF_YEAR, 1);  
+            
+            Calendar endCal = Calendar.getInstance();
+            endCal.set(Calendar.YEAR, year);
+            endCal.set(Calendar.MONTH, 11); // 11 = december
+            endCal.set(Calendar.DAY_OF_MONTH, 31);
+            
+             demands.add(Demand.builder()
+                    .consumerCode(consumerCode)
+                    .demandDetails(demandDetails)
+                    .payer(owner)
+                    .minimumAmountPayable(minimumPayableAmt)
+                    .tenantId(tenantId)
+                    .taxPeriodFrom( startCal.getTimeInMillis())
+                    .taxPeriodTo(endCal.getTimeInMillis())
+                    .consumerType("FSM")
+                    .businessService(utils.getBillingBusinessService(calculation.getFeeType()))
+                    .build());
+        }
+        return demandRepository.saveDemand(requestInfo,demands);
+    }
+
+    /**
+     * Adds roundOff taxHead if decimal values exists
+     * @param tenantId The tenantId of the demand
+     * @param demandDetails The list of demandDetail
+     */
+    private void addRoundOffTaxHead(String tenantId,List<DemandDetail> demandDetails){
+        BigDecimal totalTax = BigDecimal.ZERO;
+
+        DemandDetail prevRoundOffDemandDetail = null;
+
+        /*
+        * Sum all taxHeads except RoundOff as new roundOff will be calculated
+        * */
+        for (DemandDetail demandDetail : demandDetails){
+        	/*if(!demandDetail.getTaxHeadMasterCode().equalsIgnoreCase(BPACalculatorConstants.MDMS_ROUNDOFF_TAXHEAD))
+                totalTax = totalTax.add(demandDetail.getTaxAmount());
+            else*/
+             prevRoundOffDemandDetail = demandDetail;
+        }
+
+        BigDecimal decimalValue = totalTax.remainder(BigDecimal.ONE);
+        BigDecimal midVal = new BigDecimal(0.5);
+        BigDecimal roundOff = BigDecimal.ZERO;
+
+        /*
+        * If the decimal amount is greater than 0.5 we subtract it from 1 and put it as roundOff taxHead
+        * so as to nullify the decimal eg: If the tax is 12.64 we will add extra tax roundOff taxHead
+        * of 0.36 so that the total becomes 13
+        * */
+        if(decimalValue.compareTo(midVal) > 0)
+            roundOff = BigDecimal.ONE.subtract(decimalValue);
+
+
+        /*
+         * If the decimal amount is less than 0.5 we put negative of it as roundOff taxHead
+         * so as to nullify the decimal eg: If the tax is 12.36 we will add extra tax roundOff taxHead
+         * of -0.36 so that the total becomes 12
+         * */
+        if(decimalValue.compareTo(midVal) < 0)
```

Comment:
What happens in case of roundoff=0.5?

Reply:
In case of .5 no roundoff.

Reply:
@talele08 for 0.5 do you want to round off to upper value or lower value ?

Reply:
>=0.5 the round to next value.

Reply:
use FSM_ROUNDOFF instead of TL